core/arch/riscv64/riscv64.asm

```diff
@@ -226,8 +261,36 @@ GLOBAL_LABEL(dr_setjmp:)
  */
         DECLARE_FUNC(dr_longjmp)
 GLOBAL_LABEL(dr_longjmp:)
-/* FIXME i#3544: Not implemented */
-        ret
+        ld       t0, 0 (ARG1)
+        mv       sp, t0
+        ld       x8, 8 (ARG1)
+        ld       x9, 16 (ARG1)
+        ld       x18, 24 (ARG1)
+        ld       x19, 32 (ARG1)
+        ld       x20, 40 (ARG1)
+        ld       x21, 48 (ARG1)
+        ld       x22, 56 (ARG1)
+        ld       x23, 64 (ARG1)
+        ld       x24, 72 (ARG1)
+        ld       x25, 80 (ARG1)
+        ld       x26, 88 (ARG1)
+        ld       x27, 96 (ARG1)
+        fld      f8, 104 (ARG1)
+        fld      f9, 112 (ARG1)
+        fld      f18, 120 (ARG1)
+        fld      f19, 128 (ARG1)
+        fld      f20, 136 (ARG1)
+        fld      f21, 144 (ARG1)
+        fld      f22, 152 (ARG1)
+        fld      f23, 160 (ARG1)
+        fld      f24, 168 (ARG1)
+        fld      f25, 176 (ARG1)
+        fld      f26, 184 (ARG1)
+        fld      f27, 192 (ARG1)
+        beqz     ARG1, skip
+        addi     ARG1, ARG1, 1
+skip:
+        jalr     ra
```

Comment:
ARG2 is unused. Shouldn't that be put in the return value reg (x10/arg1)?

Reply:
See reference for longjmp: https://linux.die.net/man/3/longjmp.

Reply:
Fixed https://github.com/DynamoRIO/dynamorio/pull/6071/commits/6779b4195a5960141158905e3623fd9b9d7868fe